### DISCUSSION TITLE
Add Stars count filter

### DISCUSSION THREAD

**Discussion by kjsman** [community/community#8578] [Other Feature Feedback, Questions, & Ideas]
Example: `stars:1000` only queries codes of repositories which starred more than 1k.

This would help researchers to find specific patterns from open sources projects.

**Comment by warengonzaga** [comment]
I love the idea of it...

**Comment by DamianFekete** [comment]
Add starts count filter... or be able to sort by the number of stars.

**Comment by p0** [comment]
That's a good idea! We currently use indicators like star/watcher/fork counts (and many others) to rank and score results, so you should tend to see results from repos with more stars higher up. It would be possible to expose those in the search query syntax too. I'll take it to the team.

**Reply by wumpus** [reply]
I see that the advanced search now allegedly has the ability to filter by star count, but it only searches the names of repos. And doesn't seem to work. I was hoping to search code and then filter by star count, so I could open issues against a bunch of repos that need an API update.

**Comment by intrigus-lgtm** [comment]
I'd also be interested in filtering by star count and also fork count.
For security research it might also be interesting to filter by projects that are older than xyz days/years to filter inactive/abandoned projects.

**Comment by pluraltouch** [comment]
that is a shame. Not only one, but two reasons: a) it is basic functional requirement to sort hundreds of result by some quality rank. b) As for any result item, there is a single star number attribute, and it is simple to get, there would be no technical difficulty to implement this. 

In general neither the syntax help, neither the announcement mentions the term "sort". I am wondering how on earth the team thinks that a search functionality can be useful without sorting facility?

**Comment by Wladefant** [comment]
It is already implemented: [an example](https://github.com/search?q=stars%3A%3E100000&type=repositories)

**Reply by DamianFekete** [reply]
[HttpServletRequest stars:>10](https://github.com/search?q=HttpServletRequest+stars%3A%3E10&type=code) doesn't find any code.

**Reply by cpctaylo** [reply]
I get 
> The stars qualifier is not supported when searching code. [Try searching for repositories instead?](https://github.com/search?q=HttpServletRequest+stars%3A%3E10&type=repositories&query=HttpServletRequest+stars%3A%3E10)

**Comment by goproslowyo** [comment]
I didn't realize it supported the range pattern. Exactly what I was looking for: `stars:100..5000`

**Reply by wumpus** [reply]
But it's still only searching repo names. If you look at commits in the result, there's a warning:

> The stars qualifier is not supported when searching commits. Try searching for repositories instead?

**Comment by jin-yng** [comment]
Any way we could get this feature on code searches? I have the case where I am searching for a particular usage pattern and would like to find a well-established, production tested method.

**Reply by jin-yng** [reply]
A sort of greatest to least would be most helpful.

**Comment by zeddysoft** [comment]
Do we have any update on this yet

**Comment by thecatontheceiling** [comment]
I also would love this feature to be added

**Comment by hatlesswizard** [comment]
We are still waiting for this feature to be added

**Comment by remmons-r7** [comment]
This feature would be a huge deal for me if added. +1 :) thank you!

**Comment by GermanAizek** [comment]
Microsoft still hasn't made this feature, it seems they have completely abandoned Github as actively developed product. People have been asking since 2021, but the function has been gone for 4 years. But it really would be the most convenient function.

**Comment by shinybrightstar** [comment]
Hi Community 👋

I am able to see repositories filtered by the number of stars, as shown in this screenshot below. If you have a different experience, please comment in this discussion.

We appreciate your patience as we work toward a solution.


![image](https://github.com/user-attachments/assets/5a5f4c5d-9aea-4282-ab82-f1658f3419de)

**Reply by ouuan** [reply]
We are talking about code search, not repository search. See [above](#discussioncomment-4283487)

**Reply by shinybrightstar** [reply]
Okay. Thanks for the clarification!

**Comment by amitschendel** [comment]
+1, would really love to see this feature added

**Comment by ahmedelq** [comment]
It's 2025 and honestly, I'm surprised GitHub hasn't shipped this feature yet, especially given how straightforward it seems. 

I often find myself searching for code examples to learn from or to solve specific problems. However, with so many results, it's hard to sift through and find a production-grade codebase. That's my main motivation for adding the following features:

* **Star‑count**: Let us do something like `stars:>1337` or explicitly sort by stars. Repo search has it. Why not code search?
* **Fork‑count**: Similarly for fork something like `forks:>42`.
* **Project age filters**: A `created:<YYYY‑MM‑DD>`, `pushed:>2025-07-01`, or `age:>1y` to exclude abandoned or brand‑new repos.
* **Sort syntax**: We need explicit `sort:` (`sort:stars`, `sort:forks`, `sort:updated`, etc...) or similar syntax for result ranking.

Would love to see this prioritized and shipped ASAP.

**Comment by larrasket** [comment]
Can't believe this is not supported.

**Comment by iyunusa072-hue** [comment]
Thanks

**Comment by textyre** [comment]
Even after 3 years since AI became popular, GitHub can’t make it actually work.
